modules/samples/watertank/component/api/src/main/java/org/homeautomation/watertank/api/DeviceTypeServiceImpl.java

```diff
@@ -0,0 +1,434 @@
+            log.error(ex.getMessage(), ex);
+            return Response.status(500).entity(ex.getMessage()).build();
+        } catch (JWTClientException ex) {
+            log.error(ex.getMessage(), ex);
+            return Response.status(500).entity(ex.getMessage()).build();
+        } catch (APIManagerException ex) {
+            log.error(ex.getMessage(), ex);
+            return Response.status(500).entity(ex.getMessage()).build();
+        } catch (IOException ex) {
+            log.error(ex.getMessage(), ex);
+            return Response.status(500).entity(ex.getMessage()).build();
+        } catch (UserStoreException ex) {
+            log.error(ex.getMessage(), ex);
+            return Response.status(500).entity(ex.getMessage()).build();
+        }
+    }
+
+    /**
+     * Register device into device management service
+     *
+     * @param deviceId unique identifier for given device type instance
+     * @param name     name for the device type instance
+     * @return check whether device is installed into cdmf
+     */
+    private boolean register(String deviceId, String name) {
+        try {
+            DeviceIdentifier deviceIdentifier = new DeviceIdentifier();
+            deviceIdentifier.setId(deviceId);
+            deviceIdentifier.setType(DeviceTypeConstants.DEVICE_TYPE);
+            if (APIUtil.getDeviceManagementService().isEnrolled(deviceIdentifier)) {
+                return false;
+            }
+            Device device = new Device();
+            device.setDeviceIdentifier(deviceId);
+            EnrolmentInfo enrolmentInfo = new EnrolmentInfo();
+            enrolmentInfo.setDateOfEnrolment(new Date().getTime());
+            enrolmentInfo.setDateOfLastUpdate(new Date().getTime());
+            enrolmentInfo.setStatus(EnrolmentInfo.Status.ACTIVE);
+            enrolmentInfo.setOwnership(EnrolmentInfo.OwnerShip.BYOD);
+            device.setName(name);
+            device.setType(DeviceTypeConstants.DEVICE_TYPE);
+            enrolmentInfo.setOwner(APIUtil.getAuthenticatedUser());
+            device.setEnrolmentInfo(enrolmentInfo);
+            boolean added = APIUtil.getDeviceManagementService().enrollDevice(device);
+            if (added) {
+                APIUtil.registerApiAccessRoles(APIUtil.getAuthenticatedUser());
+            }
+            return added;
+        } catch (DeviceManagementException e) {
+            log.error(e.getMessage(), e);
+            return false;
+        }
+    }
+
+    private ZipArchive createDownloadFile(String owner, String deviceName, String sketchType)
+            throws DeviceManagementException, JWTClientException, APIManagerException,
+                   UserStoreException {
+        //create new device id
+        String deviceId = shortUUID();
+        if (apiApplicationKey == null) {
+            String applicationUsername = PrivilegedCarbonContext.getThreadLocalCarbonContext().getUserRealm()
+                    .getRealmConfiguration().getAdminUserName();
```

Comment:
When a line is broken at a non-assignment operator the break comes before the symbol. 
https://google.github.io/styleguide/javaguide.html#s4.5.1-line-wrapping-where-to-break